<details><summary>Failed to submit graph for fd013e40310d690052db4ea557fb5e18d2843484
</summary>Message:
Failed to run command: 'pbp generate from-feedstock --output /tmp/tmpvcx7pq55/fakeredis-feedstock.yaml --custom-metadata-file /tmp/tmpvcx7pq55/custom_metadata.yaml --release-name fakeredis-feedstock-4-fd013e40310d690052db4ea557fb5e18d2843484 fakeredis-feedstock/recipe', with error: 'INFO:root:Found feedstock fakeredis-feedstock at /tmp/tmpvcx7pq55/fakeredis-feedstock/recipe
INFO:root:Build/Test graph from fakeredis-feedstock
Created by sbx_user1051 on 2026-06-11 16:47:17
Command line: /opt/conda/envs/webhook-listener/bin/pbp generate from-feedstock --output /tmp/tmpvcx7pq55/fakeredis-feedstock.yaml --custom-metadata-file /tmp/tmpvcx7pq55/custom_metadata.yaml --release-name fakeredis-feedstock-4-fd013e40310d690052db4ea557fb5e18d2843484 fakeredis-feedstock/recipe

INFO:root:Found abs.yaml in /tmp/tmpvcx7pq55/fakeredis-feedstock
INFO:pbpcli.utils.common:Setting env var: ANACONDA_ROCKET_ENABLE_PY314 = True.
INFO:root:Creating graph from feedstock at: fakeredis-feedstock
INFO:pbpcli.utils.common:Setting env var: CONDA_SUBDIR = linux-64.
INFO:pbpcli.utils.expand_variants:Elapsed time for get_package_variants: 0.11 seconds
INFO:pbpcli.utils.expand_variants:Matrix vars: 4 unchanging vars:(465), changing vars (4): ['blas_impl', 'cuda_compiler_version', 'numpy', 'python']
INFO:pbpcli.utils.expand_variants:Elapsed time for preparation for render: 0.00 seconds
INFO:pbpcli.utils.common:Setting env var: CONDA_PKGS_DIRS = /tmp/conda_pkgs_64xvn_cm.
Problematic recipe:



package:
  name: fakeredis
  version: 2.36.1

source:
  url: https://pypi.org/packages/source/f/fakeredis/fakeredis-2.36.1.tar.gz
  sha256: 7240567f4e1c07a2b3d30c0fd88e5e598e948c2e25850a560727de1d1d3dc946

build:
  number: 0
  script:  -m pip install . --no-deps --no-build-isolation --ignore-installed --no-cache-dir -vvv

requirements:
  host:
    - python
    - pip
    - hatchling
  run:
    - python
    - redis-py >=4.3
    - sortedcontainers >=2.4.0,<3
- sortedcontainers >=2
- typing-extensions >=4.7

test:
  requires:
    - pip
  commands:
    - pip check
  imports:
    - fakeredis

about:
  home: https://github.com/cunla/fakeredis-py
  license_file: LICENSE
  license: BSD-3-Clause
  license_family: BSD
  summary: Fake implementation of redis API (redis-py) for testing purposes.
  description: |
    FakeRedis is a pure-Python implementation of the Redis key-value store.
    It enables running tests requiring redis server without an actual server.
    It provides enhanced versions of the redis-py Python bindings for Redis.
  dev_url: https://github.com/cunla/fakeredis-py
  doc_url: https://fakeredis.readthedocs.io

extra:
  recipe-maintainers:
    - viridianmelody

INFO:pbpcli.utils.common:Unsetting env var: CONDA_PKGS_DIRS
INFO:pbpcli.utils.common:Unsetting env var: CONDA_SUBDIR
INFO:pbpcli.utils.common:Unsetting env var: ANACONDA_ROCKET_ENABLE_PY314
Traceback (most recent call last):
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 418, in yamlize
    return yaml.load(data, Loader=StringifyNumbersLoader)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/yaml/__init__.py", line 81, in load
    return loader.get_single_data()
           ~~~~~~~~~~~~~~~~~~~~~~^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/yaml/constructor.py", line 49, in get_single_data
    node = self.get_single_node()
  File "yaml/_yaml.pyx", line 674, in yaml._yaml.CParser.get_single_node
  File "yaml/_yaml.pyx", line 688, in yaml._yaml.CParser._compose_document
  File "yaml/_yaml.pyx", line 732, in yaml._yaml.CParser._compose_node
  File "yaml/_yaml.pyx", line 848, in yaml._yaml.CParser._compose_mapping_node
  File "yaml/_yaml.pyx", line 861, in yaml._yaml.CParser._parse_next_event
yaml.parser.ParserError: while parsing a block mapping
  in "<unicode string>", line 4, column 1
did not find expected key
  in "<unicode string>", line 25, column 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/envs/webhook-listener/bin/pbp", line 10, in <module>
    sys.exit(cli())
             ~~~^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1442, in __call__
    return self.main(*args, **kwargs)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1363, in main
    rv = self.invoke(ctx)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1226, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 794, in invoke
    return callback(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 430, in wrapper
    return func(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 700, in from_feedstock
    build_configs, test_configs = expand_build_matrix(
                                  ~~~~~~~~~~~~~~~~~~~^
        feedstock.recipe_dir_path, src_channels, parsed_user_variants
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 96, in expand_build_matrix
    _targets, build_configs, test_configs = variants_for_feedstock_and_platform(
                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        recipe_dir_path, platform, src_channels, user_variants
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/utils/expand_variants.py", line 568, in variants_for_feedstock_and_platform
    metadata_tuples = api.render(
        parsed.recipe,
    ...<2 lines>...
        variants=user_variants,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/api.py", line 60, in render
    metadata_tuples = render_recipe(
        recipe_path,
    ...<4 lines>...
        permit_unsatisfiable_variants=permit_unsatisfiable_variants,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 1048, in render_recipe
    m = MetaData(str(recipe), config=config)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 1250, in __init__
    self.parse_again(permit_undefined_jinja=True, allow_no_other_outputs=True)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 1338, in parse_again
    self.meta = parse(
                ~~~~~^
        self._get_contents(
        ^^^^^^^^^^^^^^^^^^^
    ...<5 lines>...
        path=self.meta_path,
        ^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 581, in parse
    res = yamlize(data)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 422, in yamlize
    raise UnableToParse(original=e)
          ~~~~~~~~~~~~~^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/exceptions.py", line 22, in __init__
    super().__init__(*args, **kwargs)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
TypeError: CondaError.__init__() missing 1 required positional argument: 'message'
'
aws lambda request id: c7e84787-5456-56b9-87ae-2523bce9bd5a
[View in Grafana](https://grafana.anacondaconnect.com/explore?schemaVersion=1&panes=%7B%2286g%22:%7B%22datasource%22:%22loki-central%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7B__aws_cloudwatch_log_group%3D%5C%22%2Faws%2Flambda%2Fpbp-graph-submitter-prod%5C%22%7D%20%7C%20json%20%7C%20requestId%20%3D%20%60c7e84787-5456-56b9-87ae-2523bce9bd5a%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki-central%22%7D,%22editorMode%22:%22builder%22,%22direction%22:%22backward%22%7D%5D,%22range%22:%7B%22from%22:%22now-3h%22,%22to%22:%22now%22%7D,%22panelsState%22:%7B%22logs%22:%7B%22columns%22:%7B%220%22:%22Time%22,%221%22:%22Line%22%7D,%22visualisationType%22:%22table%22,%22labelFieldName%22:%22labels%22%7D%7D%7D%7D&orgId=1)</details>

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -10,7 +10,7 @@ source:
   sha256: 7240567f4e1c07a2b3d30c0fd88e5e598e948c2e25850a560727de1d1d3dc946
 
 build:
-  number: 1
+  number: 0
   script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation --ignore-installed --no-cache-dir -vvv
 
 requirements:

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -10,7 +10,7 @@ source:
   sha256: 7240567f4e1c07a2b3d30c0fd88e5e598e948c2e25850a560727de1d1d3dc946
 
 build:
-  number: 0
+  number: 1
   script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation --ignore-installed --no-cache-dir -vvv
 
 requirements:
@@ -21,9 +21,14 @@ requirements:
   run:
     - python
     - redis-py >=4.3
-    - sortedcontainers >=2.4.0,<3
-- sortedcontainers >=2
-- typing-extensions >=4.7  # [py<311]
+    - sortedcontainers >=2
+    - typing-extensions >=4.7  # [py<311]
+  run_constrained:
+    - lupa >=2.1  # lua
+    - jsonpath-ng >=1.6  # json, vectorset
+    - pyprobables >=0.6  # bf, cf, probabilistic
+    - valkey >=6  # [py>=38]  # valkey
+    - numpy >=2.4.0  # [py>=311]  # vectorset
 
 test:
   requires:

